include/dds/core/Time.hpp

```diff
@@ -129,47 +148,161 @@ class OMG_DDS_API Time
      * @return comparison result
      */
     int compare(
-            const Time& that) const;
+            const Time& that) const
+    {
+        if (*this > that)
+        {
+            return 1;
+        }
+        else if (*this < that)
+        {
+            return -1;
+        }
+        else if (*this == that)
+        {
+            return 0;
+        }
+    }
 
     /**
      * @param that Time to compare
      * @return true if the Time is greater than the comparator
      */
     bool operator >(
-            const Time& that) const;
+            const Time& that) const
+    {
+        if (this->sec() > that.sec())
+        {
+            return true;
+        }
+        else if (this->sec() < that.sec())
+        {
+            return false;
+        }
+        else
+        {
+            if (this->nanosec() > that.nanosec())
+            {
+                return true;
+            }
+            else
+            {
+                return false;
+            }
+        }
+    }
 
     /**
      * @param that Time to compare
      * @return true if the Time is greater than or equal to the comparator
      */
     bool operator >=(
-            const Time& that) const;
+            const Time& that) const
```

Comment:
Can be implemented as inline:
return !(*this < that)

It reduces code error chances keeping the logic in one place for all comparisons.